Lib/test/test_zoneinfo/test_zoneinfo.py

```diff
@@ -1720,13 +1720,35 @@ def test_env_variable_relative_paths(self):
                 with self.subTest("warning", path_var=path_var):
                     # Note: Per PEP 615 the warning is implementation-defined
                     # behavior, other implementations need not warn.
-                    with self.assertWarns(self.module.InvalidTZPathWarning):
+                    with self.assertWarns(self.module.InvalidTZPathWarning) as w:
                         self.module.reset_tzpath()
+                    self.assertEqual(w.warnings[0].filename, __file__)
 
                 tzpath = self.module.TZPATH
                 with self.subTest("filtered", path_var=path_var):
                     self.assertSequenceEqual(tzpath, expected_paths)
 
+    def test_env_variable_relative_paths_warning_location(self):
+        path_var = "path/to/somewhere"
+
+        with self.python_tzpath_context(path_var):
+            with self.subTest("module-level warning"):
+                with CleanImport("zoneinfo", "zoneinfo._tzpath"):
+                    with self.assertWarns(RuntimeWarning) as w:
+                        import zoneinfo
+                # Since we use import hacks, we cannot just use `isinstance`
+                self.assertEqual(
+                    type(w.warnings[0].message).__qualname__,
+                    "InvalidTZPathWarning",
+                )
+                # It should represent the current file:
+                self.assertEqual(w.warnings[0].filename, __file__)
+
+            with self.subTest("warning"):
+                with self.assertWarns(self.module.InvalidTZPathWarning) as w:
+                    self.module.reset_tzpath()
+                self.assertEqual(w.warnings[0].filename, __file__)
```

Comment:
Isn't it already tested in other method?

Reply:
There is already an `assertWarns()` in the other method. We only need to add the check for the filename there. And no need to duplicate it here.

But the warning in `import zoneinfo` was not tested, so in needed a new test.